lib/vmfloaty/abs.rb

```diff
@@ -258,9 +257,14 @@ def self.retrieve(verbose, os_types, token, url, user, config, _ondemand = nil)
           :user => user,
         },
       },
-      :vm_token => vmpooler_config['token'] # request with this token, on behalf of this user
     }
 
+    if config['vmpooler_config']
+      vmpooler_config = Utils.get_vmpooler_service_config(config['vmpooler_fallback'])
+      # request with this token, on behalf of this user
+      req_obj[:vm_token] = vmpooler_config['token']
```

Comment:
Should this be using the `token` method parameter? That will be the `--token abc123` value provided on the CLI.

Like, should it attempt to use `token` for this value, and if null then fallback to the vmpooler config token?

Reply:
No you cannot use the ABS token with Vmpooler, there is one token for each and not the same.

So what you have here would work, but if the vmpooler token is not included in the get, then you are not able to use bit-bar to track your VMs or use `floaty list --active --service vmpooler`